pkgs/development/tools/rye/default.nix

```
@@ -9,24 +11,40 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "rye";
-  version = "unstable-2023-04-23";
+  version = "0.5.0";
 
-  src = fetchFromGitHub {
-    owner = "mitsuhiko";
-    repo = "rye";
-    rev = "b3fe43a4e462d10784258cad03c19c9738367346";
-    hash = "sha256-q9/VUWyrP/NsuLYY1+/5teYvDJGq646WbMXptnUUUyw=";
+  src = fetchgit {
+    url = "https://github.com/mitsuhiko/${pname}.git";
+    rev = version;
+    # One of the tests runs git command so we need .git directory there
+    leaveDotGit = true;
+    sha256 = "lTE3UzmHzzRBxqLl5zngZyuF5KcCDGqxOGGUODhZAlA=";
   };
 
-  cargoHash = "sha256-eyJ6gXFVnSC1aEt5YLl5rFoa3+M73smu5wJdAN15HQM=";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "dialoguer-0.10.4" = "sha256-WDqUKOu7Y0HElpPxf2T8EpzAY3mY8sSn9lf0V0jyAFc=";
+    };
+  };
+
+  # Remove the patch after the next update: https://github.com/mitsuhiko/rye/pull/275
+  patches = [ ./offline-licenses.patch ];
+
+  env = {
+    OPENSSL_NO_VENDOR = 1;
+  };
 
   nativeBuildInputs = [ pkg-config ];
 
   buildInputs = [
+    git
     openssl
   ]
   ++ lib.optional stdenv.isDarwin SystemConfiguration;
 
+  nativeCheckInputs = [ git ];
+
```

```suggestion
  buildInputs = [
    openssl
  ]
  ++ lib.optional stdenv.isDarwin SystemConfiguration;

  nativeCheckInputs = [ git ];

```

In the case where we leave the "git test", we only need to add `git` to `nativeCheckInputs` but not to `buildInputs`.

---

`rye` depends on `git` in runtime. However it seems it is an optional dependency: https://github.com/mitsuhiko/rye/blob/11b3f8190c3e196c3c4e851c87396217004570a5/rye/src/cli/init.rs#L266

---

Oh right !
Let's keep it then :)